server/backends/blobstore/gcs/gcs.go

```diff
@@ -295,3 +297,51 @@ func (g *GCSBlobStore) Reader(ctx context.Context, blobName string) (io.ReadClos
 		return reader, nil
 	}
 }
+
+func (g *GCSBlobStore) SetBucketCustomTimeTTL(ctx context.Context, ageInDays int64) error {
+	ctx, spn := tracing.StartSpan(ctx)
+	attrs, err := g.bucketHandle.Attrs(ctx)
+	spn.End()
+	if err != nil {
+		return err
+	}
+
+	// If the lifecycle is already set correctly, return nil.
+	for _, rule := range attrs.Lifecycle.Rules {
+		if rule.Condition.DaysSinceCustomTime == ageInDays &&
+			rule.Action.Type == storage.DeleteAction {
+			return nil
+		}
+	}
+
+	// Otherwise, attempt to set the bucket lifecycle.
```

Comment:
i see how it works, but i think it's worth spelling out in a comment that sending the empty `Rules` list causes the ttl to be removed if applicable?

Reply:
Done. Also fixed a tiny bonus bug.